Updated the Forge MDK and Minecraft version to 1.18.2.

Zachary-Concordia opened the pull request:
Updated the Forge MDK to the latest stable release for minecraft version 1.18.2 so we can better use the system level testing framework. Separated from the testing since this update effects the rest of the mod and its probably better to get it out asap.  Code changes are only semantic changes for depreciated functions that were renamed. Only other change was uncommenting the particle effect on the block generation so it doesn't give purple texture not found things.

Play testing was not effected by changes.

APLab-ZJ-Projects committed:
Updated the Forge MDK and minecraft version to 1.18.2 to allow for usage of the testing framework for system level tests. Small semantic adjustments made to code to make it compatible with 1.18.2.

APLab-ZJ-Projects committed:
Updated the Forge MDK and minecraft version to 1.18.2 to allow for usage of the testing framework for system level tests. Small semantic adjustments made to code to make it compatible with 1.18.2.

APLab-ZJ-Projects committed:
Updated the Forge MDK and minecraft version to 1.18.2 to allow for usage of the testing framework for system level tests. Small semantic adjustments made to code to make it compatible with 1.18.2.

APLab-ZJ-Projects committed:
Updated the Forge MDK and minecraft version to 1.18.2 to allow for usage of the testing framework for system level tests. Small semantic adjustments made to code to make it compatible with 1.18.2.

jamestjw commented on src/main/java/com/ocelotslovebirds/birdhaus/datagen/RecipesGen.java:
```diff
@@ -31,26 +30,26 @@ public RecipesGen(DataGenerator generatorIn) {
     @Override
     protected void buildCraftingRecipes(Consumer<FinishedRecipe> consumer) {
-        birdhouseRecipe(consumer, Registration.BIRDHOUSE_BLOCK.get(), BlockTags.LOGS, BlockTags.PLANKS);
+        birdhouseRecipe(consumer, Registration.BIRDHOUSE_BLOCK.get(), Items.OAK_LOG, Items.OAK_PLANKS);
```
why are we doing this?

Zachary-Concordia replied:
This led me on an adventure! I didn't know why initially, only that it was the recommended fix. So the reason is that minecraft made 'has' a private function that no longer can unlock a recipe based on tags, so all the tag-based unlocks for data-gen broke even though the recipes themselves still work fine. I went and readded that function and will upload a version that still takes tags shortly.

jamestjw commented on build.gradle:
```diff
@@ -111,6 +144,8 @@ minecraft {
 
 repositories {
+    mavenCentral()
+
```
you introduced an extra line here

Zachary-Concordia replied:
ahhhhhhh. Thanks 👍  :)

APLab-ZJ-Projects committed:
Updated the Recipe Generator to be able to still use tags rather than having to use ItemLikes.

APLab-ZJ-Projects committed:
Updated the Recipe Generator to be able to still use tags rather than having to use ItemLikes.

APLab-ZJ-Projects committed:
Removed extra line :O

APLab-ZJ-Projects committed:
Removed extra line :O

jamestjw commented on build.gradle:
```diff
@@ -121,11 +155,21 @@ repositories {
 }
 
 dependencies {
+    implementation 'junit:junit:4.13.2'
+    implementation 'org.junit.jupiter:junit-jupiter:5.8.2'
```
btw is there a reason why we promoted `org.junit.jupiter:junit-jupiter:5.8.2` to `implementation`? Whats up with `junit:junit:4.13.2`?

Zachary-Concordia replied:
Not sure what happened with this change, I must've messed up merging from the MDK or gotten careless with IntelliJ's suggested changes. I reverted this entirely.

jamestjw commented on src/main/java/com/ocelotslovebirds/birdhaus/datagen/RecipesGen.java:
```diff
@@ -51,6 +52,10 @@ private static void birdhouseRecipe(
 
     @Override
     protected void buildCraftingRecipes(Consumer<FinishedRecipe> consumer) {
-        birdhouseRecipe(consumer, Registration.BIRDHOUSE_BLOCK.get(), BlockTags.LOGS, BlockTags.PLANKS);
+        birdhouseRecipe(consumer, Registration.BIRDHOUSE_BLOCK.get(), ItemTags.LOGS, ItemTags.PLANKS);
+    }
+
+    protected static InventoryChangeTrigger.TriggerInstance has(TagKey<Item> tagItem) {
+        return inventoryTrigger(ItemPredicate.Builder.item().of(tagItem).build());
     }
 }
```
Ok cool this works! I was going suggest the usage of reflection but this works too :)

jamestjw replied:
if I understand the diff between private and protected correctly, wouldn't this method be better off as a private one, since we don't expect any other class to invoke it

Zachary-Concordia replied:
It would have to be protected since it is overriding the RecipeProvider's protected 'has' function and you can't have an override in Java that changes access level. However I think this brings up a really good point - there isn't really a need to override the has function just to re-implement tag possession. I changed the function name to possesses instead of 'has' and made it private, that way it is more clear that the function is only in the RecipesGen class and not quietly overriding the parent's function.

jamestjw commented on build.gradle:
```diff
@@ -61,6 +79,26 @@ minecraft {
+            }
+        }
+
+        // This run config launches GameTestServer and runs all registered gametests, then exits.
+        // By default, the server will crash when no gametests are provided.
+        // The gametest system is also enabled by default for other run configs under the /test command.
+        gameTestServer {
+            workingDirectory project.file('run')
+
```
ah i see what's going on, this is from the other changes that you're making, okay I guess x.x

jamestjw committed:
Implement birdhouse bird spawning [OLB-51] (#6)

* Implement birdhouse bird spawning
- Birdhouse spawns birds every 50 ticks
- Implement ticker to manage server ticks

* Spawn bird at random offset from birdhouses

* Refactor seed consumption to use the Ticker class

* Fix seed consumption behaviour

* Set bird spawn y-offset to 10 units above a birdhouse

chris7neves committed:
added bird ai to hang around birdhouse if active (#9)

* added bird ai to hang around birdhouse if active

* fixed styling errors

* fixed small bug. Removed extraneous functions

* rebased from main, restructured goal logic to fit new ticking structure

* Styling changes, removed dead code in hangoutaroundbirdhousegoal class. Refactored goal class. Added comments

* added bird ai to hang around birdhouse if active

* fixed styling errors

* fixed small bug. Removed extraneous functions

Zachary-Concordia committed:
Update to the build.gradl to the 3-25-22 MDK, reverted implementation changes, reimplemented Parchment mappings so things make better sense. Rebased onto current version.

Zachary-Concordia committed:
Update to the build.gradl to the 3-25-22 MDK, reverted implementation changes, reimplemented Parchment mappings so things make better sense. Rebased onto current version.

jamestjw commented:
ok everything is cool, but the commit history though :o what happened

Zachary-Concordia committed:
Merge branch 'main' into mdk_update

Zachary-Concordia committed:
Merge remote-tracking branch 'origin/mdk_update' into mdk_update

Zachary-Concordia committed:
Trying to undo the commit mess I somehow made. :/
